### DISCUSSION TITLE
Non-refundable streams

### DISCUSSION THREAD

**Discussion by smol-ninja** [sablier-labs/evm-monorepo#1345] [Ideas]
RustyRabbit (from Cantina) suggested to add a feature to be able to set stream "non-refundable".

### Some use cases
- Create a non-refundable stream so that other people can participate in funding without having to trust the sender
- Gitcoin grants. Imagine Gitcoin using Flow to create streams towards grantee which can be funded by anyone. 

Starting this discussion to track similar feature requests from Flow users.

**Comment by smol-ninja** [comment]
A user has a requirement that can be solved with this feature.

<img width="400" alt="Screenshot 2024-12-19 at 18 55 20" src="https://github.com/user-attachments/assets/13c5f569-8f20-4836-9e87-63d20dd23093" />

Should we implement it in the Flow 1.2.0?

cc @sablier-labs/solidity

**Reply by PaulRBerg** [reply]
That user was strictly interested in vesting streams. Not Flow. See https://github.com/sablier-labs/v2-core/discussions/1121.
